projects/cm_mcu/LocalTasks.c

```diff
@@ -1299,10 +1299,6 @@ int init_load_clk(int clk_n)
   if (clk_n == 0)
     i2c_addrs = CLOCK_CHIP_R0A_I2C_ADDR;
 
-  // grab the semaphore to ensure unique access to I2C controller
-  while (xSemaphoreTake(clock_args.xSem, (TickType_t)10) == pdFALSE)
-    ;
-
```

Comment:
is it right that the issue here is that the early returns for bad EEPROM were done w/o releasing the semaphore, so that is why you had to move the scope of the semaphore take/release?

Reply:
yes, we have discussed this and the sem give/take is moved to a larger scope.